sdk/monitor/monitor_client.go

```diff
@@ -32,6 +37,11 @@ type client struct {
 	cancel context.CancelFunc
 }
 
+type recvResult struct {
```

Comment:
Let's make that type `recvResult` function-scoped, since it is used only inside `NewClient`?

Reply:
>Let's make that type recvResult function-scoped, since it is used only inside NewClient?

Can you describe this? Is it one of the go patterns? Will this clear for reading?

Reply:
I think if you have ability to reduce the scope of your type, better to do it